clang/tools/clang-sycl-linker/ClangSYCLLinker.cpp

```diff
@@ -474,9 +518,11 @@ int main(int argc, char **argv) {
   DryRun = Args.hasArg(OPT_dry_run);
   SaveTemps = Args.hasArg(OPT_save_temps);
 
-  OutputFile = "a.spv";
-  if (Args.hasArg(OPT_o))
-    OutputFile = Args.getLastArgValue(OPT_o);
+  IsAOTCompileNeeded = Args.hasArg(OPT_arch);
```

Comment:
I removed the `--target` option to align with @asudarsa's PR; now AOT compilation is performed when `-arch` is supplied.